data_pipeline/generic_web_api/generic_web_api_data_etl.py

```diff
@@ -449,6 +478,43 @@ def process_web_api_data_etl_batch_with_batch_source_value(
         LOGGER.info('completed batch: %d', 1 + batch_index)
 
 
+def process_web_api_data_etl_batch_with_batch_source_value(
+    data_config: WebApiConfig,
+    all_source_values_iterator: Optional[Iterable[dict]] = None,
+    batch_source_value: Optional[dict] = None
+):
+    stored_state = get_start_timestamp_from_state_file_or_optional_default_value(
```

Comment:
As a future improvement it might be easier to understand if we used a function that provided date ranges, rather than calling the next function in the chain.